src/Nethermind/Nethermind.Serialization.Rlp/BlockDecoder.cs

```diff
@@ -49,15 +51,31 @@ public class BlockDecoder : IRlpValueDecoder<Block>, IRlpStreamDecoder<Block>
 
             rlpStream.Check(unclesCheck);
 
+            List<Withdrawal> withdrawals = null;
+
+            if (header.Timestamp >= HeaderDecoder.WithdrawalTimestamp)
+            {
+                int withdrawalsLength = rlpStream.ReadSequenceLength();
+                int withdrawalsCheck = rlpStream.Position + withdrawalsLength;
+                withdrawals = new();
```

Comment:
So we could do the same for BlockDecoder and try to initialize some capacity based on transactions RLP length. For withdrawals it will be more deterministic. However, I think it is out of the scope a bit.